kasa/discover.py

```diff
@@ -297,16 +316,17 @@ async def discover_single(
         finally:
             transport.close()
 
-        if host in protocol.discovered_devices:
-            dev = protocol.discovered_devices[host]
+        if ip in protocol.discovered_devices:
+            dev = protocol.discovered_devices[ip]
+            dev.host = host
```

Comment:
Should we rather keep the discovered host as-is?

Reply:
By keeping setting the host as the hostname rather than the ip returned via UDP we are keeping the behaviour consistent with the way discover_single worked in 0.5.3 whereby there would be no translation from a hostname into an ip.  Also the use case we are meeting here is where users are adding devices by hostname to HA.  In order for that to work as expected the device `host` needs to be the hostname rather than the ip.